### PR TITLE
support ENI

### PR DESCRIPTION
**Pull request by nightmeng**
TF_DEBUG=debug TF_ACC=1 go  test -v -run=TestAccAlicloudNetworkInterface
=== RUN   TestAccAlicloudNetworkInterface_importBasic
--- PASS: TestAccAlicloudNetworkInterface_importBasic (31.00s)
=== RUN   TestAccAlicloudNetworkInterfaceBasic
--- PASS: TestAccAlicloudNetworkInterfaceBasic (36.39s)
=== RUN   TestAccAlicloudNetworkInterfaceWithPrivateIpList
--- PASS: TestAccAlicloudNetworkInterfaceWithPrivateIpList (47.63s)
=== RUN   TestAccAlicloudNetworkInterfaceWithPrivateIpCount
--- PASS: TestAccAlicloudNetworkInterfaceWithPrivateIpCount (43.08s)
=== RUN   TestAccAlicloudNetworkInterfaceWithoutPrimaryIpAddress
--- PASS: TestAccAlicloudNetworkInterfaceWithoutPrimaryIpAddress (31.90s)
PASS
ok  	github.com/terraform-providers/terraform-provider-alicloud/alicloud	190.038s

### CODE REVIEW

File: alicloud/resource_alicloud_network_interface.go [hunk 1]
```diff
@@ -0,0 +1,342 @@
...
+	return resourceAliyunNetworkInterfaceUpdate(d, meta)
+}
+
+func resourceAliyunNetworkInterfaceRead(d *schema.ResourceData, meta interface{}) error {
+	client := meta.(*connectivity.AliyunClient)
+	ecsService := EcsService{client}
+
+	eni, err := ecsService.DescribeNetworkInterfaceById("", d.Id())
+	if err != nil {
+		return fmt.Errorf("Describe NetworkInterface(%s) failed, %#v", d.Id(), err)
```
Comment (xiaozhu36) [inline alicloud/resource_alicloud_network_interface.go]: If there is no any enis, please return nil

File: alicloud/resource_alicloud_network_interface_test.go [hunk 2]
```diff
@@ -0,0 +1,544 @@
...
+					resource.TestCheckResourceAttrSet("alicloud_network_interface.eni", "private_ip"),
+					resource.TestCheckResourceAttr("alicloud_network_interface.eni", "private_ips.#", "0"),
+					resource.TestCheckResourceAttr("alicloud_network_interface.eni", "description", "No primary private IP address test"),
+					resource.TestCheckResourceAttr("alicloud_network_interface.eni", "tags.TF-VER", "0.11.3"),
+				),
+			},
+		},
+	})
+}
+
```
Comment (xiaozhu36) [inline alicloud/resource_alicloud_network_interface_test.go]: This is missing the resource exist method

File: alicloud/resource_alicloud_network_interface_test.go [hunk 3]
```diff
@@ -0,0 +1,544 @@
...
+		return fmt.Errorf("No ENI ID is set")
+	}
+
+	client := testAccProvider.Meta().(*connectivity.AliyunClient)
+	ecsService := EcsService{client}
+	_, err := ecsService.DescribeNetworkInterfaceById("", rs.Primary.ID)
+	if err != nil {
+		if NotFoundError(err) {
+			return nil
+		}
```
Comment (xiaozhu36) [inline alicloud/resource_alicloud_network_interface_test.go]: If the resource has been deleted , please `continue` not `return nil`

File: alicloud/resource_alicloud_network_interface_test.go [hunk 4]
```diff
@@ -0,0 +1,544 @@
...
+    "available_resource_creation"= "VSwitch"
+}
+
+resource "alicloud_vswitch" "vswitch" {
+    name = "tf-testAcc-vswitch"
+    count = 1
+    cidr_block = "192.168.0.0/24"
+    availability_zone = "${data.alicloud_zones.default.zones.0.id}"
+    depends_on = [ "alicloud_vpc.vpc" ]
+    vpc_id = "${alicloud_vpc.vpc.id}"
```
Comment (xiaozhu36) [inline alicloud/resource_alicloud_network_interface_test.go]: The line 304 and 307 is redundant and please remove it

File: alicloud/resource_alicloud_network_interface_test.go [hunk 5]
```diff
@@ -0,0 +1,544 @@
...
+	"time"
+
+	"github.com/aliyun/alibaba-cloud-sdk-go/services/ecs"
+	"github.com/hashicorp/terraform/helper/resource"
+	"github.com/hashicorp/terraform/terraform"
+	"github.com/terraform-providers/terraform-provider-alicloud/alicloud/connectivity"
+)
+
+func init() {
+	resource.AddTestSweepers("alicloud_network_interface", &resource.Sweeper{
```
Comment (xiaozhu36) [inline alicloud/resource_alicloud_network_interface_test.go]: I think this resource sweeper test should be dependence of vswitch, security group

File: alicloud/resource_alicloud_network_interface.go [hunk 6]
```diff
@@ -0,0 +1,342 @@
...
+	}
+
+	d.Set("name", eni.NetworkInterfaceName)
+	d.Set("description", eni.Description)
+	d.Set("vswitch_id", eni.VSwitchId)
+	d.Set("private_ip", eni.PrivateIpAddress)
+	d.Set("security_groups", eni.SecurityGroupIds.SecurityGroupId)
+	d.Set("status", eni.Status)
+	privateIps := make([]string, 0, len(eni.PrivateIpSets.PrivateIpSet))
```
[the remark this answers is not in the document]
Comment (nightmeng) [reply alicloud/resource_alicloud_network_interface.go]: Got it

File: alicloud/resource_alicloud_network_interface.go (hunk 1 above)
Comment (nightmeng) [reply alicloud/resource_alicloud_network_interface.go]: Got it

File: alicloud/resource_alicloud_network_interface_test.go (hunk 2 above)
Comment (nightmeng) [reply alicloud/resource_alicloud_network_interface_test.go]: Got it

File: alicloud/resource_alicloud_network_interface_test.go (hunk 3 above)
Comment (nightmeng) [reply alicloud/resource_alicloud_network_interface_test.go]: Got it

**Pull request merged**